engine/access/rpc/engine.go

```diff
@@ -177,7 +177,22 @@ func (e *Engine) OnFinalizedBlock(block *model.Block) {
 // No errors expected during normal operations.
 func (e *Engine) processOnFinalizedBlock(_ *model.Block) error {
 	finalizedHeader := e.finalizedHeaderCache.Get()
-	return e.backend.ProcessFinalizedBlockHeight(finalizedHeader.Height)
+
+	var err error
+	// NOTE: The BlockTracker is currently only used by the access node and not by the observer node.
+	if e.backend.BlockTracker != nil {
```

Comment:
Can we call `ProcessOnFinalizedBlock` everytime but perform the if statement inside? This makes more sense to me than checking if the `BlockTracker` is available and then processing finalized block.